source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -945,6 +960,38 @@ void InjaTransformer::transform(Http::RequestOrResponseHeaderMap &header_map,
   } else if (merged_extractors_to_body_) {
     std::string output = json_body.dump();
     maybe_body.emplace(output);
+  } else if (!merge_templates_.empty()) {
+
+    for (const auto &merge_template : merge_templates_) {
+      const std::string &name = std::get<0>(merge_template);
+      
+      // prepare variables for non-advanced_templates_ scenario
+      absl::string_view name_to_split;
+      json* current = nullptr;
+      // if (!advanced_templates_) {
+      name_to_split = name;
+      current = &json_body;
```

Comment:
is the intent to mutate what is in   typed_tls_data.context_?
Or should this be a copy?

I think the answer is that it should mutate whats in the context_ field but I figure I should at least ask

Reply:
This is a copy/paste of the logic for extractors, but that is a very very good point, any operations done on the body will be impacted by this. I think it's a very valid question of whether this should happen before or after the rest of the transformations. Extractors directly modify the JSON body beforehand, so maybe we should do the same?